### DISCUSSION TITLE
How does yolov8 model reduce the misidentification of similar objects? What is the reasonable proportion of background images

### DISCUSSION THREAD

**Discussion by hhhFuture** [ultralytics/ultralytics#19909] [Solutions]
When yolov8 trains the object detection model, how much background should be included to prevent misidentification? For example, if I have 6w pictures of scissors hands, the model will misrecognize the hand or three-finger gesture during training. How much background picture should I add? The background image mentioned here is an image that does not have scissor hands but contains a palm or other gesture.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @hhhFuture, thank you for your question and for using Ultralytics 🚀! 

We recommend referring to the [Docs](https://docs.ultralytics.com/) for comprehensive guidance, including [training tips](https://docs.ultralytics.com/guides/model-training-tips/) that can help optimize your results. Specifically, the section on dataset preparation may provide valuable insights for balancing background images and object classes in your dataset.

If this is a custom training ❓ question, we encourage you to share more details about your setup, including:
- Dataset distribution (e.g., number of background images vs. labeled images)
- Examples of dataset images
- Training parameters and logs

This will help the community and Ultralytics engineers provide more targeted advice.

If you're experiencing a 🐛 bug, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can assist you in debugging.

💡 Additionally, you can engage with the wider Ultralytics community for insights:
- Join our [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time discussions.
- Visit our [Discourse forum](https://community.ultralytics.com/) for detailed conversations.
- Share on our [Subreddit](https://reddit.com/r/Ultralytics) to learn from others' experiences.

## Upgrade

Ensure you're using the latest `ultralytics` package with all dependencies. Upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can train YOLO models in any of these verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests validate YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

This is an automated response, but an Ultralytics engineer will follow up with you soon to provide additional assistance 😊.

**Comment by glenn-jocher** [comment]
@hhhFuture for YOLOv8's improved distinction between similar objects, we recommend focusing on dataset quality rather than quantity. Include 10-30% background images containing challenging negatives like palms/gestures without scissor hands. The anchor-free split head and advanced architecture in YOLOv8 help reduce false positives - see our Training documentation for dataset best practices. For your specific case, ensure your background images include the exact hand gestures causing confusion to create hard negative examples.

**Reply by hhhFuture** [reply]
Thank you for your reply. I have read the code of hyperparameter Mosaic. The code is to splices the original image randomly, and then resize it to the size of 640x640. Therefore, in the training, some targets of the image have been cropped, or half of the targets have been cropped. For example, scissor gestures, after using Mosaic data enhancement, some scissor gestures are cut and only the fist part is left. I know the Mosaic technique can increase the generalization of the model, but does the Mosaic technique lead to false positives? For example, I mentioned the scissor gesture above, two fingers were cut out, leaving only the fist part, resulting in the model misidentifying the fist gesture

**Reply by glenn-jocher** [reply]
@hhhFuture thanks for your thoughtful question! Mosaic augmentation can indeed create partial objects during training, but this intentionally helps models learn to recognize objects from partial patterns while maintaining context awareness. For scissor gestures specifically, we recommend: 1) Ensuring your dataset contains sufficient full-object examples, and 2) Gradually reducing the `mosaic` probability in later training stages (e.g., `hyp.mosaic=0.5`) to balance partial/complete samples. The Configuration documentation explains how to adjust augmentation hyperparameters for optimal results.

**Reply by hhhFuture** [reply]
Thanks for your advice, I did not have misrecognition after adding the background image, but the recognition effect of the model was relatively poor, but the mAP after the training was quite high, and I fell into the confusion. I wonder if it's overfitting, or if the added background image makes it bad at recognizing. My data set is based on more than 50,000 original pictures, taking 50% at random, and then rotating ±90° respectively to get a total of more than 120,000 pictures, among which the background pictures added (the background pictures do not include my scissors gesture, they are all three fingers or hands, ok gesture and other pictures) are 12,000. The way I trained was to first train 50 rounds with the data without background graphs, and then train 50 rounds on the basis of last.pt, at which time the data set had 12,000 background graphs added. In addition, I found a problem. After 30 rounds of training based on last.pt, I used the 30-round model for reasoning, and the model almost did not misidentify, and the generalization and recognition effect were quite good. After 40 rounds of training, the generalization ability of the model was reduced a lot after mosaic data enhancement was turned off. Finally, the model recognition effect becomes worse after the training is completed. I am very confused now, I hope you can give me some advice, thank you very much! The picture below is a screenshot 
![F1_curve](https://github.com/user-attachments/assets/c5257cf6-aaaf-41ea-bf4f-7cbda5eb79b8)
of my training
![训练结束后的截图](https://github.com/user-attachm
![labels](https://github.com/user-attachments/assets/bda65277-b8b0-478b-8089-e2f8887d6002)
![confusion_matrix](https://github.com/user-attachments/assets/490e0f46-4ff0-4963-a1e7-6aa39a7bfd11)

ents/assets/5a71a57e-2732-4d2b-9a41-e305a7d892c6)
![results](https://github.com/user-attachments/assets/1a306641-73a5-4f2c-9b5c-38bf98e3f5ed)

**Reply by glenn-jocher** [reply]
Thank you for sharing these details. The behavior you're describing suggests potential overfitting when adding background images and modifying augmentation schedules. Based on your F1 curve and confusion matrix, consider these adjustments: 

1. **Gradual Mosaic Reduction**: Instead of fully disabling mosaic augmentation later in training, decay the `mosaic` probability gradually using a cosine scheduler in your hyperparameters file (see our Augmentation Settings documentation for implementation guidance)

2. **Validation Set Composition**: Ensure your validation set contains representative samples with and without background images to monitor generalization

3. **Learning Rate Adjustment**: When resuming training with new data (background images), reduce your initial learning rate by 50-70% to avoid overwriting previously learned features

These strategies typically help maintain generalization while incorporating challenging negative samples. Let us know if you need clarification on implementing any of these adjustments.

**Reply by hhhFuture** [reply]
Thank you for your suggestion. When the parameters are not changed, I only reduce the data of the background image by 50%, and the model effect is much better, positive recognition is improved a lot, and misrecognition is reduced a lot, but some tricky gestures still cause misrecognition. I want to modify it according to your suggestion in your latest reply, but I don't know how to modify it, and I don't know how to gradually reduce mosaic in the later stage, only see the close_mosaic parameter, the function of which is as follows: Disable Mosaic data enhancement for the last N epodes to stabilize training before training is complete. Set to 0 disables this feature. The default parameter is 10. The following is my training code, please help to see how to gradually reduce the Mosaic, how to adjust the learning rate: When using new data (background image) to resume training, reduce the initial learning rate by 50-70% to avoid overwriting the previously learned features
from ultralytics import YOLO

if __name__ == '__main__':
    model = YOLO(r"E:\Project_Gesture\model_script\runs\detect\gesture_scissors8\weights\last.pt")
    model.train(
        data=r"E:\Project_Gesture\model_script\scissors.yaml",
        imgsz=640,
        device=0,
        epochs=50,
        batch=24,
        name="gesture_scissors",
        fliplr=0.5,  
        flipud=0.5,  
        degrees=15,  
        mosaic=1.0,  
        mixup=0.3,  
        plots=True  
    )

**Reply by glenn-jocher** [reply]
To implement gradual mosaic reduction and learning rate adjustment in your training script, modify your code like this:

```python
model.train(
    data=r"...",
    epochs=50,
    lr0=0.003,  # Reduced by 70% from default 0.01
    mosaic=0.5,  # Start with 50% mosaic probability
    close_mosaic=20,  # Disable mosaic completely in last 20 epochs
    ...
)
```

This creates a stepped reduction pattern while maintaining stability. For true gradual decay, you would need to modify the hyperparameters file's mosaic scheduling logic directly. The Ultralytics Train Settings documentation explains advanced configuration of augmentation schedules.
